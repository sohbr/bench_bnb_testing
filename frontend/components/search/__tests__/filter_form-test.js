import React from "react";
import FilterForm from "../filter_form";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json"; //for snapshot testing

const setup = () => {
  const props = {
    minSeating: 2,
    maxSeating: 5,
    updateFilter: jest.fn()
  };

  const filterFormWrapper = shallow(<FilterForm {...props} />);

  return {
    filterFormWrapper,
    props
  };
};

describe("FilterForm", () => {
  let minSeatFilter;
  const { filterFormWrapper, props } = setup();
  minSeatFilter = filterFormWrapper.find("input").first();

  test("minSeating filter defaults to the passed in props of minSeating", () => {
    // your code here
    expect(minSeatFilter.prop("value")).toEqual(props.minSeating);
  });

  test("updates filter onChange", () => {
    // Your code here
    minSeatFilter.props().onChange({ currentTarget: { value: 3 } });
    expect(props.updateFilter.mock.calls.length).toBe(1);
  });

  // test("renders correctly", () => {
  //   // Snapshot testing here
  // });
});
