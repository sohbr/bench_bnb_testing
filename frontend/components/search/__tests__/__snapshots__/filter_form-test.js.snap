// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterForm renders correctly 1`] = `
<div>
  <span
    className="filter"
  >
    Filter results:
  </span>
  <br />
  <label>
    Minimum Seats
  </label>
  <input
    onChange={[Function]}
    type="number"
    value={2}
  />
  <br />
  <label>
    Maximum Seats
  </label>
  <input
    onChange={[Function]}
    type="number"
    value={5}
  />
</div>
`;
